frontend/src/app/modules/global_search/global-search-input.component.ts

```diff
@@ -0,0 +1,338 @@
+import {GlobalSearchService} from "app/modules/global_search/global-search.service";
+import {debounceTime, distinctUntilChanged} from "rxjs/operators";
+import {untilComponentDestroyed} from "ng2-rx-componentdestroyed";
+import {CurrentProjectService} from "app/components/projects/current-project.service";
+import {Subject} from "rxjs";
+
+export const globalSearchSelector = 'global-search-input';
+
+@Component({
+  selector: globalSearchSelector,
+  templateUrl: './global-search-input.component.html'
+})
+
+export class GlobalSearchInputComponent implements OnDestroy {
+  @ViewChild('inputEl') input:ElementRef;
+  @ViewChild('btn') btn:ElementRef;
+
+  public focused:boolean = false;
+  public noResults = false;
+  public searchTerm:string = '';
+  public expanded:boolean = false;
+
+  private searchTermChanged:Subject<string> = new Subject<string>();
+
+  private $element:JQuery;
+  private $input:JQuery;
+
+  private unregisterGlobalListener:Function | undefined;
+
+  public text:{ [key:string]:string } = {
+    all_projects: this.I18n.t('js.global_search.all_projects'),
+    this_project: this.I18n.t('js.global_search.this_project'),
+    this_project_and_all_descendants: this.I18n.t('js.global_search.this_project_and_all_descendants'),
+    search: this.I18n.t('js.global_search.search') + ' ...'
+  };
+
+  constructor(readonly FocusHelper:FocusHelperService,
+              readonly elementRef:ElementRef,
+              readonly I18n:I18nService,
+              readonly PathHelperService:PathHelperService,
+              readonly halResourceService:HalResourceService,
+              readonly dynamicCssService:DynamicCssService,
+              readonly globalSearchService:GlobalSearchService,
+              readonly cdRef:ChangeDetectorRef,
+              readonly currentProjectService:CurrentProjectService) {
+  }
+
+  private projectScopeTypes = ['all_projects', 'this_project', 'this_project_and_all_descendants'];
+
+  ngOnInit() {
+    this.$element = jQuery(this.elementRef.nativeElement);
+    this.$input = jQuery(this.input.nativeElement);
+
+    this.searchTermChanged
+      .pipe(
+        distinctUntilChanged(),
+        debounceTime(250),
+        untilComponentDestroyed(this)
+      )
+      .subscribe((searchTerm:string) => {
+        this.searchTerm = searchTerm;
+        this.determineExpansion();
+        this.cdRef.detectChanges();
+      });
+
+    this.globalSearchService.searchTerm$
+      .pipe(
+        distinctUntilChanged(),
+        untilComponentDestroyed(this)
+      )
+      .subscribe((searchTerm:string) => {
+        this.searchTerm = searchTerm;
+        this.determineExpansion();
+        this.cdRef.detectChanges();
+      });
+
+    let selected = false;
+
+    this.$input.autocomplete({
+      delay: 250,
+      autoFocus: true,
+      appendTo: '#top-menu',
+      classes: {
+        'ui-autocomplete': 'search-autocomplete--results'
+      },
+      position: {
+        my: 'left top+9',
```

Comment:
What is `+9`? Is this a typo or functionality?

Reply:
This is syntax of jQuery UI that allows you to position what they call `Widgets`. Took me quite some time to find that :-)